### PR TITLE
feat: account watcher e2e

### PR DESCRIPTION
**Pull request by k-g-j**
## **Description**

This PR introduces end-to-end (E2E) tests for the account watcher feature. The tests cover the following scenarios:

- Adding a watch account using a valid externally owned address (EOA).
- Ensuring the “Send,” “Swap,” and “Bridge” buttons are disabled for watch accounts.
- Handling various invalid inputs, such as missing prefixes, invalid addresses, and ENS names.
- Preventing the import of private keys for watched addresses.
- Verifying the absence of the “Show private key” button for watch accounts.
- Removing and re-adding a watch account to validate proper account management.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26524?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/438

## **Manual testing steps**

1. Build for test: `yarn build:test`
2. Run the new e2e: `yarn test:e2e:single test/e2e/accounts/create-watch-account.spec.ts --browser=chrome --leave-running`

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/accounts/create-watch-account.spec.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  await driver.clickElement('[data-testid="account-menu-icon"]');
+  await driver.clickElement(
+    '.multichain-account-list-item--selected [data-testid="account-list-item-menu-button"]',
+  );
+  await driver.clickElement('[data-testid="account-list-menu-remove"]');
+  await driver.clickElement({ text: 'Remove', tag: 'button' });
+}
+
+describe('Account-watcher snap', function (this: Suite) {
+  it('user can add watch account with valid EOA address', async function () {
```
Comment (gantunesr) [inline test/e2e/accounts/create-watch-account.spec.ts:69]:
```suggestion
  it('adds watch account with valid EOA address', async function () {
```

Comment (k-g-j) [reply test/e2e/accounts/create-watch-account.spec.ts]: fixed here [166e2fb](https://github.com/MetaMask/metamask-extension/pull/26524/commits/166e2fbfd99d1fa11cf404a7f1ce5070a037c99f)

**Review by owencraston — DISMISSED** [review]

File: test/e2e/accounts/create-watch-account.spec.ts
Comment (gantunesr) [inline test/e2e/accounts/create-watch-account.spec.ts:1]: Lets add one more test where you disable and then re-enable the toggle and the option to add a wath-only behaves as expected

Comment (k-g-j) [reply test/e2e/accounts/create-watch-account.spec.ts:1]: here you go [16dfd85](https://github.com/MetaMask/metamask-extension/pull/26524/commits/16dfd8593ee51a2296c40ce069d3e66800c60553)

**Review by owencraston — APPROVED** [review]

**Review by gantunesr — APPROVED** [review]
Good work

**Pull request merged**